admin/notifiers/class-subscription-notifier.php

```diff
@@ -0,0 +1,416 @@
+	 *
+	 * @return Yoast_Notification_Center
+	 *
+	 * @codeCoverageIgnore
+	 */
+	protected function get_notification_center() {
+		return Yoast_Notification_Center::get();
+	}
+
+	/**
+	 * Initializes the subscription notification.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->current_notification_id = get_option( self::NOTIFICATION_ID );
+
+		$this->subscription_with_earliest_expiry_date = $this->determine_subscription_with_earliest_expiry_date();
+
+		if ( $this->subscription_with_earliest_expiry_date === null ) {
+			$this->clean_up();
+			return;
+		}
+
+		$days_until_expiration = $this->calculate_days_until_expiration( $this->subscription_with_earliest_expiry_date );
+
+		$this->determine_notification( $days_until_expiration );
+	}
+
+	/**
+	 * Calculates the number of days until a subscription expires. 0 or below means the subscription has expired.
+	 *
+	 * @param stdClass $subscription Subscription to get the subscription expiration time for.
+	 *
+	 * @return int Number of days until expiration.
+	 */
+	protected function calculate_days_until_expiration( $subscription ) {
+		return (int) ceil( ( strtotime( $subscription->expiry_date ) - time() ) / DAY_IN_SECONDS );
+	}
+
+	/**
+	 * Sets the current notification to one of the defined constants, based on a set of conditions.
+	 *
+	 * @param int $days_until_expiration Days until the subscription expires.
+	 *
+	 * @return void
+	 */
+	private function determine_notification( $days_until_expiration ) {
+		if ( $days_until_expiration < 28 && $days_until_expiration > 7 ) {
+			$this->set_current_notification( self::EXPIRATION_WITHIN_4_WEEKS );
+
+			/* translators: %1$s expands to Yoast, %2$s expands to an opening anchor tag, %3$s expands to a percentage, %4$s expands to an closing anchor tag. */
+			$message = sprintf( __( 'Your %1$s plugin(s) will expire within 4 weeks. When plugins expire, you will no longer receive updates or support. %2$sRenew now to get a %3$s discount!%4$s', 'wordpress-seo' ), 'Yoast', '<a href="' . $this->get_url() . '"">', '25%', '</a>' );
+			$this->show_notification( $message );
+			return;
+		}
+		if ( $days_until_expiration <= 7 && $days_until_expiration > 1 ) {
+			$this->set_current_notification( self::EXPIRATION_WITHIN_1_WEEK );
+
+			$formatted_date = '<b>' . date_i18n( 'F jS, HH:MM', strtotime( $this->subscription_with_earliest_expiry_date->expiry_date ) ) . '</b>';
+			/* translators: %1$s expands to Yoast, %2$s expands to a formatted date and time (e.g. "Februari 8th 15:00"), %3$s expands to a percentage, %4$s expands to an opening anchor tag, %5$s expands to an closing anchor tag. */
+			$message        = sprintf( esc_html__( 'Your %1$s plugins are about to expire! When plugins expire, you will no longer receive updates or support. You have until %2$s to renew with a %3%s discount. %4$sRenew now!%5$s', 'wordpress-seo' ), 'Yoast', $formatted_date, '25%', '<a href="' . $this->get_url() . '">', '</a>' );
```

Comment:
Typo `%3%s` should be `%3$s`
<img width="271" alt="Screenshot 2019-04-01 at 11 27 39" src="https://user-images.githubusercontent.com/1682452/55317643-37759080-5471-11e9-82ae-8edd7181bebd.png">